lib/hydra/works/services/generic_file/add_file_to_generic_file.rb

```diff
@@ -50,13 +48,40 @@ def persist
           end
         end
 
-        def attach_attributes(path, original_name, mime_type)
-          current_file.content = ::File.open(path)
-          current_file.original_name = original_name ? original_name : ::File.basename(path)
-          current_file.mime_type = mime_type ? mime_type : Hydra::PCDM::GetMimeTypeForFile.call(path)
+        def attach_attributes(file) 
+          current_file.content = file
+          current_file.original_name = determine_original_name(file)
+          current_file.mime_type = determine_mime_type(file)
+        end
+
+        # Return mime_type based on methods available to file
+        # @param object for mimetype to be determined. Attempts to use methods: :mime_type, :content_type, and :path.
+        def determine_mime_type(file)
```

Comment:
Remove the explicit `return` from lines 61, 63, 65, and 67 and rely upon implicit returns.